src/Presentation/UiBuilder.php

```diff
@@ -105,16 +101,12 @@ private function buildFacetsViewModel( ?ItemId $itemType, Query $query ): array
 
 	private function buildFacetViewModel( FacetConfig $facet, PropertyConstraints $state ): array {
 		return [
-			'label' => $this->getPropertyLabel( $facet->propertyId ),
+			'label' => $this->facetLabelBuilder->getFacetLabel( $facet->propertyId ),
```

Comment:
I think the prior code is better.

Services normally cater to one use case. It's weird to have multiple public methods, as `FacetLabelBuilder` now has. 

The previous property label code works, is simple, and is tested. Perhaps the little logic it contains can be de-duplicated if you create a generic improved version of `LabelLookup` for other parts of the code. That generic improved version "BetterLabelLookup" could do the fallback to the ID serialization and return `string ` instead of `Term`. Unless you are already creating such an interface anyway for other code, it does not seem worth it to touch the existing code here.

Reply:
Suggestion: drop `getFacetLabel` from `facetLabelBuilder`